src/lib/checkpoint/cp_mgr.cpp

```diff
@@ -69,24 +71,29 @@ void CPManager::register_consumer(cp_consumer_t consumer_id, std::unique_ptr< CP
     }
 }
 
+[[nodiscard]] CPGuard CPManager::cp_guard() { return CPGuard{}; }
```

Comment:
Why isn't this just defined in the header where CPGuard(); is also declared public? Also, can this be static or at least `const`? It seems to use no instance fields.

Reply:
CPManager all other methods are singleton. I can understand it seems weird. CPManager is a singleton and we need to call the current instance of cp_manager() cp_io_enter(). So I have 2 options
1) Make this `const`, but it is technically not a const, because CPGuard calls cp_io_enter() which is not a const
2) CPGuard constructor will pass CPManager non const pointer as well and it maintains inside it, (so copy constructor can use that CPManager instance).

Reply:
I don't think I follow this pattern...how can it matter where `CPGuard()` is called from (inside method of CPManager or not)? There is nothing passed to it to link it back to the instance that called it that I see?

Reply:
I did the option 2) above, so now it always pass (this) pointer from CPManager.